runtime/src/iree/hal/drivers/hip/hip_device.c

```diff
@@ -1200,44 +1432,60 @@ static iree_status_t iree_hal_hip_device_queue_execute(
     iree_hal_command_buffer_t* command_buffer,
     iree_hal_buffer_binding_table_t binding_table) {
   iree_hal_hip_device_t* device = iree_hal_hip_device_cast(base_device);
-
   IREE_TRACE_ZONE_BEGIN(z0);
+
+  if (queue_affinity == IREE_HAL_QUEUE_AFFINITY_ANY) {
+    queue_affinity = 0x1;
+  }
+
+  uint64_t queue_affinity_mask =
+      ((iree_hal_queue_affinity_t)1 << device->topology.count);
+  queue_affinity_mask = queue_affinity_mask | (queue_affinity_mask - 1);
+  queue_affinity &= queue_affinity_mask;
+
+  uint64_t nidx = iree_math_count_trailing_zeros_u64(queue_affinity);
```

Comment:
another new novel name :) use device_ordinal, queue_ordinal, etc consistently to make it clear what's being referenced